### ISSUE TITLE
thegamedb API has changed?

### ISSUE THREAD

**Issue by pgiblock** [sselph/scraper#230]
Not sure, but I am a first time user of this scraper. I ran into the issue "It appears that thegamesdb.net isn't up". Looking at the code, it appears that the scraper attempts to GET `http://thegamesdb.net/api/GetGame.php?id=1`. After following the 302, a 404 is returned. From https://api.gamesdb.net , it appears the API has changed? Looks like one now needs to hit `https://api.thegamesdb.net/Games/ByGameID?id=1&apikey=<API_KEY>`.

Is this a recent change on gdb's side? Are there any plans to support the new API?

**Comment by sselph** [comment]
Thanks for the heads-up. It looks like it has changed. I'll convert the
code over this weekend.

**Comment by sselph** [comment]
Actually I may have to remove support for this service. The apiKey they mention is for the dev and is limited to something like 1000 queries per month. I think they designed the API quota for people running a web server or something that mirrors the data and not a scraper like mine.

Ideally they would reconsider and allow users to generate an API key and they'd use their own individual quota. The shared quota for an app like mine makes no sense.

**Comment by pgiblock** [comment]
Yeah. Batching sounds ideal to get the query count down. I haven't dug into the guts of the scraper enough to know how painful of a refactor that would be.

**Comment by pgiblock** [comment]
Good news: It seems that simply replace 'thegamedb.net' with 'legacy.thegamedb.net' is a usable stop-gap solution.

**Comment by sselph** [comment]
Nice.

Yeah the code today is my first Go code so not great to start and over the years has grown to become even less elegant. It does something roughly like the following so not laid out for batch processing in a single database.

for each rom found
  for each DB:
     if result:
       break
     else:
       continue

 It was more designed to try multiple databases to fill in gaps from that were missing. A refactor would probably need to do something like
for each DB:
  for each batch of unscraped roms:
    get results(batch)

**Comment by pgiblock** [comment]
Yeah, that makes sense, where `unscraped roms` is initially the full set. Then for each iteration of DB, it is only the set of unresolved roms from the previous iteration.

**Comment by sselph** [comment]
Thanks. Once I get an API key, I'll start working on it more seriously but if you have documentation of the response formats I can go ahead and have most of it ready. I'll start looking at refactoring the code to make batching a little easier since the new API seems to encourage that.

**Comment by Zer0xFF** [comment]
im afraid thats not available yet, as there are still few more things to implement, and they take priority over documentation.

and we hope that keys will be reissued by next weekend.

**Comment by symbios24** [comment]
Hi, upon the change of the api it finds very little game images per system eg for nes in 400 roms it finds 200 for gameboy in 250 roms it finds 100 is this going to be fixed?

**Comment by symbios24** [comment]
after updating the scraper the xml files has the same address : thegamesdb.net instead of the legacy.thegamesdb.net is this normal?

**Comment by Zer0xFF** [comment]
@symbios24 the legacy subdomain is the old site with only the domain change, so results returned shouldn't be any difference.

I changed any references I was able to find, but there is the possibility I missed some, which endpoint is still returning thegamesdb.net?

**Comment by symbios24** [comment]
so far i tried the gameboy/nes/atari 2600 games and they have the thegamesdb.net to the xml

**Comment by symbios24** [comment]
also atari 5200 is still returning thegamesdb.net i assume all the atari systems do the same

**Comment by sselph** [comment]
Thanks for the report. I may have forgotten to fix a url somewhere. I'll
also see if there was some change affecting images. I would expect them all
to work or to not work so it seems weird that it is hit or miss.

**Comment by symbios24** [comment]
Another thing is the ps1-psx  *pbp* file extensions if barely finds any images is it possible to improve/fix this?

**Comment by symbios24** [comment]
if you can change the scraper for pbp - psx files to download images/pictures based on the name of the game and not on the extension of the filename will be great.

**Comment by danger89** [comment]
It will require that this project (scraper) will request a API key, [see this post](https://forums.thegamesdb.net/viewtopic.php?t=14).

So you can use the new API, eg: https://api.thegamesdb.net/#/Games/GamesByGameName
